python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,306 @@
+    """
+
+    raise_on_fail: bool = False
+    parquet_options: ParquetOptions = dataclasses.field(default_factory=ParquetOptions)
+    executor: StreamingExecutor | InMemoryExecutor = dataclasses.field(
+        default_factory=InMemoryExecutor
+    )
+    device: int | None = None
+
+    @classmethod
+    def from_polars_engine(cls, engine: pl.GPUEngine) -> Self:
+        """
+        Create a `ConfigOptions` object from a `pl.GPUEngine` object.
+
+        This creates our internal, typed, configuration object from the
+        user-provided `polars.GPUEngine` object.
+        """
+        # these are the valid top-level keys in the engine.config that
+        # the user passes as **kwargs to GPUEngine.
+        valid_options = {
+            "executor",
+            "executor_options",
+            "parquet_options",
+            "raise_on_fail",
+        }
+
+        extra_options = set(engine.config.keys()) - valid_options
+        if extra_options:
+            raise TypeError(f"Unsupported executor_options: {extra_options}")
+
+        user_executor = engine.config.get("executor", "in-memory")
+        if user_executor is None:
+            user_executor = "in-memory"
+        # we mutate this below
+        user_executor_options = dict(engine.config.get("executor_options", {}))
+        user_parquet_options = engine.config.get("parquet_options", {})
+        user_raise_on_fail = engine.config.get("raise_on_fail", False)
+
+        # These are user-provided options, so we need to actually validate
+        # them.
+
+        if user_executor not in {"in-memory", "streaming"}:
+            raise ValueError(f"Unknown executor '{user_executor}'")
+
+        if not isinstance(user_raise_on_fail, bool):
+            raise TypeError("GPUEngine option 'raise_on_fail' must be a boolean.")
+
+        executor: InMemoryExecutor | StreamingExecutor
+
+        match user_executor:
+            case "in-memory":
+                executor = InMemoryExecutor(**user_executor_options)
+            case "streaming":
+                # backwards compatibility for the old parquet_blocksize option
+                if (
+                    "parquet_blocksize" in user_executor_options
+                    and "parquet_partition_blocksize" in user_executor_options
+                ):
+                    raise ValueError(
+                        "Specify just one of 'parquet_blocksize' or 'parquet_partition_blocksize'."
+                    )
+                elif "parquet_blocksize" in user_executor_options:
+                    warnings.warn(
+                        "The 'parquet_blocksize' option is deprecated. Use 'parquet_partition_blocksize' instead.",
+                        FutureWarning,
+                        stacklevel=3,
+                    )
+                    user_executor_options["parquet_partition_blocksize"] = (
+                        user_executor_options.pop("parquet_blocksize")
+                    )
```

Comment:
There are no actual users of the streaming executor yet, so I don't think there is any great reason to add a deprecation warning.

With that said, I'm personally -1 on the new name anyway. It make the option name longer, and we don't gain much (if anything) from the change. I also think it's outside the scope of this PR to change the specific option names (unless something is fundamentally wrong/broken).

I think the goal of this change is to clarify that this is related to streaming partition size, and to distinguish it from the general `parquet_options`. Can we just remove this option in favor of a general `target_partition_size` in a follow-up. The docs can say that supported IO functions (only parquet at first) will use this as a guideline for partitioning.